lib/mqtt.c

```diff
@@ -146,15 +146,15 @@ static int mqtt_getsock(struct Curl_easy *data,
 /* Add the username flag to the CONNECT Variable Header*/
 static char user_flag(char flag)
 {
-  char usr_flag = 0x80;
+  char usr_flag = (char)0x80;
```

Comment:
I have my doubts on this line, because 0x80 correspond to 128, and I think that might be in the edge of a char. According to these articles, it seems to depend on the architecture.
1. [Range of char type value in C](https://stackoverflow.com/questions/4647764/range-of-char-type-values-in-c)
2. [Range of char Values](https://docs.microsoft.com/en-us/cpp/c-language/range-of-char-values?view=msvc-160#:~:text=All%20signed%20character%20values%20range,range%20from%200%20to%20255.)

Any suggestion will be appreciated.